src/php/ext/grpc/channel.c

```diff
@@ -303,6 +304,7 @@ PHP_METHOD(Channel, __construct) {
       channel->wrapper = le->channel;
     }
   }
+  gpr_mu_unlock(&channel->wrapper->mu);
```

Comment:
Sorry, I should clarify.  I think you need a separate mutex specifically for ```persistent_list```.  The linked list is not threadsafe, so any time we access it we need to acquire a global mutex around it (separate from the channel mutex).